src/lib/wallet/PaymentLinks.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import logger from '../../lib/logger/pino-logger'
+import Config from '../../config/config'
+import { generateShareLink } from '../share'
+import { GoodWallet as GoodWalletClass } from './GoodWalletClass'
+
+const log = logger.child({ from: 'GoodWallet' })
+if (Config.contractsVersion >= '1.0.0') {
+  const defaultPromiEvents: PromiEvents = {
+    onTransactionHash: () => {},
+    onReceipt: () => {},
+    onConfirmation: () => {},
+    onError: () => {},
+  }
+  log.debug('patching GoodWallet with new payment links methods')
+
+  /**
+   * deposits the specified amount to _oneTimeLink_ contract and generates a link that will send the user to a URL to withdraw it
+   * @param {number} amount - amount of money to send using OTP
+   * @param {string} reason - optional reason for sending the payment (comment)
+   * @param {({ link: string, code: string }) => () => any} getOnTxHash - a callback that returns onTransactionHashHandler based on generated code
+   * @param {PromiEvents} events - used to subscribe to onTransactionHash event
+   * @returns {{code, hashedCode, paymentLink}}
+   */
+  GoodWalletClass.prototype.generateLink = function(
```

Comment:
to avoid GoodWalletClass.prototype.xxx = function on each override and use syntax similar to class declaration you could do:

```
Object.extend(GoodWalletClass.prototype, {
  generateLink(
    amount: number,
    reason: string = '',
    events: PromiEvents = defaultPromiEvents
  ) {
    //
  },

  getWithdrawLink(otlCode: string) {
    //
  },

 // other overrides
});

Reply:
ok. this will probably be removed in one of the next versions.